dao-services/src/main/java/org/exoplatform/wiki/jpa/EntityConverter.java

```diff
@@ -184,70 +189,128 @@ public static PageEntity convertPageToPageEntity(Page page, WikiDAO wikiDAO) {
     return permissionEntities;
   }
 
-  public static Attachment convertAttachmentEntityToAttachment(AttachmentEntity attachmentEntity) {
+  public static Attachment convertAttachmentEntityToAttachment(FileService fileService, AttachmentEntity attachmentEntity) {
     Attachment attachment = null;
+    FileItem fileItem = null;
     if (attachmentEntity != null) {
       attachment = new Attachment();
-      attachment.setName(attachmentEntity.getName());
-      attachment.setTitle(attachmentEntity.getTitle());
       attachment.setFullTitle(attachmentEntity.getFullTitle());
-      attachment.setCreator(attachmentEntity.getCreator());
       if (attachmentEntity.getCreatedDate() != null) {
         Calendar createdDate = Calendar.getInstance();
         createdDate.setTime(attachmentEntity.getCreatedDate());
         attachment.setCreatedDate(createdDate);
       }
-      if (attachmentEntity.getUpdatedDate() != null) {
-        Calendar updatedDate = Calendar.getInstance();
-        updatedDate.setTime(attachmentEntity.getUpdatedDate());
-        attachment.setUpdatedDate(updatedDate);
+      try {
+        fileItem = fileService.getFile(attachmentEntity.getAttachmentFileID());
+      } catch (FileStorageException e) {
+        return attachment;
```

Comment:
try/catch should not control the flow IMO. Here we should only log an error.
If fileItem is null it will go to the end of the method and return attachment.